Networking/Networking/Remote/ProductsRemote.swift

```diff
@@ -222,7 +224,15 @@ public final class ProductsRemote: Remote, ProductsRemoteProtocol {
                                      parameters: parameters,
                                      availableAsRESTRequest: true)
         let mapper = ProductListMapper(siteID: siteID)
-        return try await enqueue(request, mapper: mapper)
+
+        let (products, responseHeaders) = try await enqueueWithResponseHeaders(request, mapper: mapper)
+
+        // Extracts the total number of pages from the response headers.
+        // Response header names are case insensitive.
+        let totalPages = responseHeaders?.first(where: { $0.key.lowercased() == Remote.PaginationHeaderKey.totalPagesCount.lowercased() })
+            .flatMap { Int($0.value) }
```

Comment:
Hm good question, I was mostly afraid that some plugins or hosts (for site credentials login with direct site requests) might alter the headers and remove the pagination headers. In this case where the pagination headers are missing, I'm not sure if it's worth blocking the merchants from using POS because of the missing headers. WDYT?

Reply:
I think it depends... 
1. What complexity does any workaround add to the code? (If we have to keep the previous workaround in, that's a strong sign we shouldn't do it.)
2. What kind of experience will we give them if the headers aren't present? (If it's just going to keep trying to load the next page, that's not great either, especially if those calls are rapidly repeated which is a risk here.)
3. How likely is it that the headers aren't present?

[The docs for v3 of the API say:](https://woocommerce.github.io/woocommerce-rest-api-docs/#pagination)
> The total number of resources and pages are always included in the X-WP-Total and X-WP-TotalPages HTTP headers.
[Which is repeated back to v1 of the API](https://woocommerce.github.io/woocommerce-rest-api-docs/wp-api-v1.html#pagination) – WC 2.6 got that, and we don't support that version. 

I _think_ we can rely on them being there, but perhaps we could add some tracking now for any request that comes back without the headers, if you have concerns?

Reply:
> What complexity does any workaround add to the code? (If we have to keep the previous workaround in, that's a strong sign we shouldn't do it.)

Great points on the various considerations and tradeoffs. I think the current workaround is not very complicated even though not ideal - it's mostly in the [remote `loadProductsForPointOfSale` returning true](https://github.com/woocommerce/woocommerce-ios/blob/cc139f83809727c06f69264306f252cd1957b1d9/Networking/Networking/Remote/ProductsRemote.swift#L234) for `hasMorePages` by default, and [`PointOfSaleProductService` returning `false` when the page items are empty](https://github.com/woocommerce/woocommerce-ios/blob/cc139f83809727c06f69264306f252cd1957b1d9/Yosemite/Yosemite/PointOfSale/PointOfSaleProductService.swift#L49-L51) (there's room for optimization there to return false for items count < page size). Having dealt with all kinds of alterations in the API response by third-party plugins, I don't know about the likelihood but there's a chance some plugin or hosting provider strips some response headers. Of course there could be none, we can add tracking to the response headers missing case and observe. If it's almost zero, let's move to throw an error when response headers are missing to simplify the code. I created an issue to track this https://github.com/woocommerce/woocommerce-ios/issues/14737.

> What kind of experience will we give them if the headers aren't present? (If it's just going to keep trying to load the next page, that's not great either, especially if those calls are rapidly repeated which is a risk here.)

There should be up to 1 extra API request per load session with the current fallback implementation (before the next reload), when we fix the infinite scroll issue. If we implement the optimization to return false for response items count < page size, then the probability of extra API request is lower (only if the last page has items size = page size).